personalities/sctfe/storage.go

```diff
@@ -78,3 +87,65 @@ func (cts *CTStorage) AddIssuerChain(ctx context.Context, chain []*x509.Certific
 	}
 	return nil
 }
+
+// cachedIssuerStorage wraps an IssuerStorage, and keeps a copy the sha256 of certs it contains.
+//
+// This is intended to make querying faster. It does not keep a copy of the certs, only sha256.
+// Only up to N keys will be stored locally.
+// TODO(phboneff): add monitoring for the number of keys
+type cachedIssuerStorage struct {
+	m map[string]bool
```

Comment:
If you choose to continue using `map`, you can save a few bytes and a bit of ambiguity here/below by defining as `m map[string]struct{}` (the value takes no space like that), then below you mark existence with `m[key] = struct{}{}`